### DISCUSSION TITLE
calcGrains seems to miss some grain boundaries which are above the set threshold

### DISCUSSION THREAD

**Discussion by lchechik1** [mtex-toolbox/mtex#1166] [Ask Anything]
Hi everyone,
I was working on some EBSD data and had some questions about calcGrain.

Looking at the colour of the IPF map, there seemed to be a grain boundary missing after running calcGrains with a threshold of 10degrees. In the first image, I think the largest grain should be split in two. Calculating the misorientation between two pixels either side of the boundary (second image), I got an angle of 11.38 degrees, above the threshold of 10 - yet calcGrains puts them into the same grain.
Below is the code used to plot the figure and to calculate the misorientation of the angles.

Any input would be greatly appreciated as we're currently a bit lost as to what's happening
Thanks,
Lova

ebsd = EBSD.load(filename);
[grains,ebsd.grainId,ebsd.mis2mean] = calcGrains(ebsd,'threshold',10*degree,'boundary','tight');

figure()
plot(ebsd('Ni-superalloy'),ebsd('Ni-superalloy').orientations,'coordinates','on');hold on
plot(grains.boundary)

ori1 = ebsd(137214).orientations;
ori2 = ebsd(137828).orientations;
angle(ori1, ori2) ./ degree

![image](https://user-images.githubusercontent.com/40566588/128507823-2df7746b-5868-405c-8bd5-50baddee72c3.png)
![image](https://user-images.githubusercontent.com/40566588/128507941-20b172b1-adae-4c73-86c1-f2a4c5a392f6.png)

**Comment by nyyssont** [comment]
I suspect there is a region of low or gradually increasing misorientation that connects these areas. The regular calcGrains algorithm then combines them into a single grain. You could try creating and plotting subgrain boundaries to see if this is the case:

https://mtex-toolbox.github.io/SubGrainBoundaries.html

**Reply by lchechik1** [reply]
So adding these subgrain boundaries does catch that boundary.
However, there are more (below), which have a 22deg misorientation and are still being counted as a single grain with no subgrain...
In the image, you can see subgrains (calculated using 1deg) in blue - code below
![image](https://user-images.githubusercontent.com/40566588/128675097-1cd430aa-41b2-4c70-b749-5090b9707309.png)

[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[1*degree, 10*degree]);
grains = smooth(grains,5)

plot(ebsd('indexed'),ebsd('indexed').orientations,'figSize','large')
hold on
plot(grains.boundary,'linewidth',2)
alpha = grains.innerBoundary.misorientation.angle / (1*degree);
plot(grains.innerBoundary,'linewidth',1.5,'edgeAlpha',alpha,'linecolor','b');
hold off

**Reply by kilir** [reply]
Hi, in mtex a grain is fully enclosed by a grainboundary, anything else goes by innerBoundary. A grain may contain closed/open boundaries below the threshold used to define a grain - low angle boundaries or subgrsinboundaries and open boundaries with angles >= the theshold angles used to define agrain boundary. 
I would say that is what you see in your image.
Cheers,
Rüdiger

**Reply by lchechik1** [reply]
Hi Rudiger,
So I understand that if you have a non-complete boundary (> threshold), this will come up as a subgrain rather than a grain.
In the figure above, subgrains are plotted in blue.
If you look at the two data-tips in the top right, these are in the same grain apparently, calculating the angle between them, you get 23deg, so would expect at least a blue (subgrain) boundary there.

**Comment by zmichels** [comment]
Hmmm… it looks like the two points you are comparing are not (maybe?) directly neighboring points. If that 22º is spread across multiple steps, then you might not get a boundary identified there. It could be something else, but I have never had MTEX miss a neighbor-neighbor boundary. To confirm either way, you could compute your boundaries with a very low angle threshold and then plot all the outer and inner boundaries and color them by the disorientation angle… that would give you a good idea of what the magnitude of misorientation is across each discrete feature you see.

Zach
